src/metrics/sifis.rs

```diff
@@ -106,6 +106,94 @@ pub(crate) fn sifis_quantized(
+        Complexity::Cognitive => space.metrics.cognitive.cognitive_sum(),
+    };
+    let sum = covs
+        .iter()
+        .enumerate()
+        .try_fold(0., |acc, (i, line)| -> Result<f64> {
+            // Check if the line is null
+            let is_null = if is_covdir {
+                line.as_i64().ok_or(Error::ConversionError())? == -1
+            } else {
+                line.is_null()
+            };
+            let sum;
+            if !is_null && i >= space.start_line - 1 && i < space.end_line {
+                // If the line is not null and is covered (cov>0) the add the complexity  to the sum
+                let cov = line.as_u64().ok_or(Error::ConversionError())?;
+                if cov > 0 {
+                    sum = acc + comp;
+                } else {
+                    sum = acc;
+                }
+            } else {
+                sum = acc;
+            }
+            Ok(sum)
+        })?;
+    Ok((sum / ploc, sum))
+}
+
+// Calculate the SIFIS quantized value for a function
+// Return the value in case of success and an specif error in case of fails
+// If the complexity of the block/file is 0 the value if sifis quantized is the coverage of the file
+pub(crate) fn sifis_quantized_function(
+    space: &FuncSpace,
+    covs: &[Value],
+    metric: Complexity,
+    is_covdir: bool,
+) -> Result<(f64, f64)> {
+    let ploc = space.metrics.loc.ploc();
+    let sum =
+    //For each line find the minimum space and get complexity value then sum 1 if comp>threshold  else sum 1
+        covs.iter()
+            .enumerate()
+            .try_fold(0., |acc, (i, line)| -> Result<f64> {
+                // Check if the line is null
+                let is_null = if is_covdir {
+                    line.as_i64().ok_or(Error::ConversionError())? == -1
+                } else {
+                    line.is_null()
+                };
+                let sum;
+                if !is_null && i>= space.start_line-1 && i< space.end_line {
+                    // Get line
+                    let cov = line.as_u64().ok_or(Error::ConversionError())?;
+                    if cov > 0 {
+                        // If the line is covered get the space of the line and then check if the complexity is below the threshold
+                        let min_space: FuncSpace = get_min_space(space, i);
+                        let comp = match metric {
+                            Complexity::Cyclomatic => min_space.metrics.cyclomatic.cyclomatic(),
+                            Complexity::Cognitive => min_space.metrics.cognitive.cognitive(),
+                        };
+                        if comp > THRESHOLD {
+                            sum = acc + 2.;
+                        } else {
+                            sum = acc + 1.;
+                        }
+                    } else {
+                        sum = acc;
+                    }
+                } else {
+                    sum = acc;
+                }
```

Comment:
Question

Is it possible to simplify this code in some way? Reducing nested `if`s?

Reply:
Regrettably I did not find a way to reduce the number of ifs.